operator/ingress/v1/types.go

```diff
@@ -0,0 +1,110 @@
+package v1
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// DNSRecord is a DNS record managed in the zones defined by
+// dns.config.openshift.io/cluster .spec.publicZone and .spec.privateZone.
+//
+// Cluster admin manipulation of this resource is not supported. This resource
+// is only for internal communication of OpenShift operators.
+type DNSRecord struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the dnsecord.
+	Spec DNSRecordSpec `json:"spec,omitempty"`
+	// status is the most recently observed status of the dnsRecord.
+	Status DNSRecordStatus `json:"status,omitempty"`
+}
+
+// dnsRecordSpec are the details of a DNS record.
+type DNSRecordSpec struct {
+	// dnsName is the hostname of the DNS record
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	DNSName string `json:"dnsName,omitempty"`
+	// targets are DNS resource record targets.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	Targets []string `json:"targets,omitempty"`
+	// recordType is a resource record type.
+	//
+	// The currently supported values are A and CNAME.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	RecordType DNSRecordType `json:"recordType,omitempty"`
+	// ttl is the resource record TTL in seconds. The default is 30.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	RecordTTL int64 `json:"recordTTL,omitempty"`
+}
+
+// dnsRecordStatus is the more recently observed status of the record.
+type DNSRecordStatus struct {
+	// zones are the status of the record per zone.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Zones []DNSZoneStatus `json:"zones,omitempty"`
+}
+
+// dnsZoneStatus is the status of a record within a specific zone.
+type DNSZoneStatus struct {
+	// dnsZone is the zone where the record is expected to be managed.
```

Comment:
Omit "expected to be"?  The thing that publishes status is the thing that manages the record.  Alternatively, maybe change "managed" → "published" because the thing that manages the record may have failed to publish it.